kong/plugins/pre-function/_handler.lua

```diff
@@ -1,14 +1,43 @@
+local resty_mlcache = require "resty.mlcache"
 local sandbox = require "kong.tools.sandbox"
 local kong_meta = require "kong.meta"
 
 -- handler file for both the pre-function and post-function plugin
 
+local CONF_SENSITIVE = {
+  pg_password = true,
+  pg_ro_password = true,
+  cassandra_password = true,
+  proxy_server = true,
+}
 
 local config_cache do
 
   local no_op = function() end
 
-  local sandbox_opts = { env = { kong = kong, ngx = ngx } }
+  local shm_name = "kong_db_cache"
+  local cache = resty_mlcache.new(shm_name, shm_name, { lru_size = 1e4 })
+  local sandbox_kong = setmetatable({}, {
```

Comment:
if config is already anonymized:
```suggestion
  local sandbox_kong = setmetatable({ cache = cache }, { __index = kong })
```

Reply:
it was not, but the PDK exposes a function to get the redacted configuration